### PR TITLE
Update Configuration Cache docs to cover new approach to input tracking

### PR DESCRIPTION
**Pull request by mlopatkin**
Mandatory input declaration for system properties, environment variables
and gradle properties is gone. Only some access patterns are
discouraged.

This PR also covers input list in the configuration cache report.

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Comment by mlopatkin** [comment]
Link to the rendered doc: https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/47940320:id/distributions/gradle-7.4-docs.zip!/gradle-7.4-20211209201047%2B0000/docs/userguide/configuration_cache.html

**Review by eskatos — APPROVED** [review]
💯

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 1]
```diff
@@ -934,6 +901,19 @@ See link:{gradle-issues}13506[gradle/gradle#13506].
...
+=== Reading a subset of system properties or environment variables at configuration time
+
+Currently, it isn't possible to read a subset of system properties or environment variables at configuration time in a way that doesn't make all properties or variables build configuration inputs.
+
+See link:{gradle-issues}13819[gradle/gradle#13819].
+
+[[config_cache:not_yet_implemented:fine_grained_tracking_of_gradle_properties]]
+=== Fine-grained tracking of Gradle properties as build configuration inputs
+
+Currently, all external sources of Gradle properties (`gradle.properties` in project directories and in the `GRADLE_USER_HOME`, environment variables that set properties and properties specified with
```
Comment (bamboo) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:914]: _... environment variables **and system properties** that set properties ..._

**Review by bamboo — APPROVED** [review]

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**